### DISCUSSION TITLE
initial compile error, resource file versions?

### DISCUSSION THREAD

**Discussion by RogueRocketGames** [o3de/o3de#1890] [Help Wanted] [answered]
downloaded src as zip vs git fork & sync.

after a couple hours of compilation, several errors that looks like this happen, can't find information on how to resolve:

  MaterialEditor.Window.vcxproj -> C:\Users\[username]\o3de-development\build\windows_vs2019\lib\profile\MaterialEditor.Wi
  ndow.lib
C:\Users\[username]\o3de-development\Code\Tools\ProjectManager\Resources\ProjectManager.rc(1): error RC2175: resource file
 ProjectManager-Icon.ico is not in 3.00 format [C:\Users\[username]\o3de-development\build\windows_vs2019\Code\Tools\Proje
ctManager\ProjectManager.vcxproj]

C:\Users\[username]\o3de-development\Code\Tools\Standalone\Source\Editor\LuaEditor.rc(3): error RC2175: resource file hex_
lua.ico is not in 3.00 format [C:\Users\[username]\o3de-development\build\windows_vs2019\Code\Tools\Standalone\LuaIDE.vcxp
roj]

  ToolsCrashHandler.vcxproj -> C:\Users\[username]\o3de-development\build\windows_vs2019\lib\profile\ToolsCrashHandler.lib
C:\Users\[username]\o3de-development\Code\Tools\AssetBundler\source\AssetBundlerBatch.rc(1): error RC2175: resource file A
ssetBundlerBatch.ico is not in 3.00 format [C:\Users\[username]\o3de-development\build\windows_vs2019\Code\Tools\AssetBund
ler\AssetBundlerBatch.vcxproj]

  AssetBundler.Tests.vcxproj -> C:\Users\[username]\o3de-development\build\windows_vs2019\bin\profile\AssetBundler.Tests.e
  xe
  Automatic MOC for target MaterialEditor
C:\Users\[username]\o3de-development\Code\Framework\AzQtComponents\AzQtComponents\Gallery\Gallery.rc(1): error RC2175: res
ource file Gallery.ico is not in 3.00 format [C:\Users\[username]\o3de-development\build\windows_vs2019\Code\Framework\AzQ
tComponents\O3DEQtControlGallery.vcxproj]

  CrySystem.Static.vcxproj -> C:\Users\[username]\o3de-development\build\windows_vs2019\lib\profile\CrySystem.Static.lib
  Automatic MOC for target ToolsCrashUploader
C:\Users\[username]\o3de-development\Code\Tools\AssetBundler\source\AssetBundler.rc(1): error RC2175: resource file ui\sty
le\AssetBundler-Icon-256x256@x2.ico is not in 3.00 format [C:\Users\[username]\o3de-development\build\windows_vs2019\Code\
Tools\AssetBundler\AssetBundler.vcxproj]

C:\Users\[username]\o3de-development\Code\Tools\AssetProcessor\AssetBuilder\AssetBuilder.rc(1): error RC2175: resource fil
e AssetBuilder.ico is not in 3.00 format [C:\Users\[username]\o3de-development\build\windows_vs2019\Code\Tools\AssetProces
sor\AssetBuilder\AssetBuilder.vcxproj]

C:\Users\[username]\o3de-development\Code\Tools\Standalone\Source\Editor\ProfilerEditor.rc(3): error RC2175: resource file
 profile_icon.ico is not in 3.00 format [C:\Users\[username]\o3de-development\build\windows_vs2019\Code\Tools\Standalone\P
rofiler.vcxproj]

  CrashReporting.Uploader.vcxproj -> C:\Users\[username]\o3de-development\build\windows_vs2019\bin\profile\CrashReporting.
  Uploader.exe
  ToolsCrashUploader.vcxproj -> C:\Users\[username]\o3de-development\build\windows_vs2019\bin\profile\ToolsCrashUploader.e
  xe
  PythonBindingsExample.Tests.vcxproj -> C:\Users\[username]\o3de-development\build\windows_vs2019\bin\profile\PythonBindi
  ngsExample.Tests.exe
C:\Users\[username]\o3de-development\Gems\Atom\Tools\ShaderManagementConsole\Code\Source\Platform\Windows\ShaderManagement
Console.rc(1): error RC2175: resource file ShaderManagementConsole.ico is not in 3.00 format [C:\Users\[username]\o3de-dev
elopment\build\windows_vs2019\External\Atom-2aacef01\Tools\ShaderManagementConsole\Code\ShaderManagementConsole.vcxproj
]

  PythonBindingsExample.vcxproj -> C:\Users\[username]\o3de-development\build\windows_vs2019\bin\profile\PythonBindingsExa
  mple.exe
C:\Users\[username]\o3de-development\Gems\Atom\Tools\MaterialEditor\Code\Source\Platform\Windows\MaterialEditor.rc(1): err
or RC2175: resource file MaterialEditor.ico is not in 3.00 format [C:\Users\[username]\o3de-development\build\windows_vs20
19\External\Atom-2aacef01\Tools\MaterialEditor\Code\MaterialEditor.vcxproj]

  CrySystem.vcxproj -> C:\Users\[username]\o3de-development\build\windows_vs2019\bin\profile\CrySystem.dll

**Comment by amznestebanpapp** [comment, marked as answer]
That looks like git LFS did not pull those files properly.
Can you check if any of those files, e.g. `ProjectManager-Icon.ico` are actual files? you can open them with a text editor, if they have text (like a hash), that is file pointer to a LFS that hasn't been pulled.
If that is the case, do a `git lfs pull` and authenticate with your github repo token (not your password).

**Reply by AMZN-alexpete** [reply]
Adding that if you download the source code as the zip from the releases page it will **not** include the LFS files - you must clone the repository per the instructions in the docs here:  https://o3de.org/docs/welcome-guide/setup/setup-from-github/

Also, when you use git lfs pull you will not need to authenticate, that will only be required for pushing (uploading) LFS files.  I'm working on updating the docs for this.

We'll eventually have an installer version you can use if you prefer not to use git.

**Reply by RogueRocketGames** [reply]
yep this turned out to be the issue, i realized AFTER I posted this LOL.   I had gotten too used to downloading ZIPs of repos to try things out before investing in full git or hg repo setups. 

Resolving this did get me compiled and i have the launcher up, but now i have a different problem which is trying to launch the editor silently fails after putting up the initial application launching dialog.  trying to find documentation on how to examine any logs for issues, or further set up steps i've missed.

this is all i see, and the last status message I see is "loading plugins" before the process just ends.

![image](https://user-images.githubusercontent.com/7526926/124828138-b947a480-df2b-11eb-84ba-87f01888a385.png)

**Reply by amznestebanpapp** [reply]
Likely is missing what project should be pointing at.
There are two ways to pass a project:
1) registering it and setting it as global: `scripts\o3de.bat set-global-project`
2) passing it to the editor with the `--project-path` parameter
Try executing the Editor with `Editor.exe --project-path=AutomatedTesting`, that should open the engine with the project that is used for testing.

**Reply by RogueRocketGames** [reply]
i was opening editor from the launcher app... I'll try the CLI method above.
but i did manage to find the logs, and here's some of the entries that seem relevant:

<14:05:12> [Error] (RPI Utils) - Failed to get asset id for shader [shaders/reflections/reflectionprobeblendweight.azshader]
<14:05:12> [Error] (ReflectionProbeFeatureProcessor) - Failed to find asset for shader [shaders/reflections/reflectionprobeblendweight.azshader]
<14:05:12> (System) - 

.... resulting finally (but i'm unsure if directly causal to...)

<14:05:12> (System) - Trace::Assert
 C:\Users\username\o3de\Code\Framework\AzCore\AzCore/std/smart_ptr/intrusive_ptr.h(189): (22064) 'class AZ::RPI::Shader *__cdecl AZStd::intrusive_ptr<class AZ::RPI::Shader>::operator ->(void) const'
<14:05:12> (System) - You can't dereference a null pointer
<14:05:12> (System) - ------------------------------------------------
<14:05:12> (System) - C:\Users\username\o3de\Gems\Atom\Feature\Common\Code\Source\ReflectionProbe\ReflectionProbeFeatureProcessor.cpp (67) : AZ::Render::ReflectionProbeFeatureProcessor::Activate
<14:05:12> (System) - C:\Users\username\o3de\Gems\Atom\RPI\Code\Source\RPI.Public\Scene.cpp (129) : AZ::RPI::Scene::Activate
<14:05:12> (System) - C:\Users\username\o3de\Gems\Atom\Bootstrap\Code\Source\BootstrapSystemComponent.cpp (285) : AZ::Render::Bootstrap::BootstrapSystemComponent::GetOrCreateAtomSceneFromAzScene
<14:05:12> (System) - C:\Users\username\o3de\Gems\Atom\Tools\AtomToolsFramework\Code\Source\Viewport\RenderViewportWidget.cpp (126) : <lambda_db90d0b5c20f29ad13cbd849ddc4705a>::operator()
<14:05:12> (System) - C:\Users\username\o3de\Code\Framework\AzCore\AzCore\EBus\Internal\BusContainer.h (1540) : AZ::Internal::EBusContainer<AZ::Render::Bootstrap::Request,AZ::Render::Bootstrap::Request,0,0>::Dispatcher<AZ::EBus<AZ::Render::Bootstrap::Request,AZ::Render::Bootst
<14:05:12> (System) - C:\Users\username\o3de\Gems\Atom\Tools\AtomToolsFramework\Code\Source\Viewport\RenderViewportWidget.cpp (135) : AtomToolsFramework::RenderViewportWidget::SetScene
<14:05:12> (System) - C:\Users\username\o3de\Code\Editor\EditorViewportWidget.cpp (3026) : EditorViewportWidget::UpdateScene
<14:05:12> (System) - C:\Users\username\o3de\Code\Editor\EditorViewportWidget.cpp (1468) : EditorViewportWidget::SetViewportId
<14:05:12> (System) - C:\Users\username\o3de\Code\Editor\ViewPane.cpp (306) : CLayoutViewPane::AttachViewport
<14:05:12> (System) - C:\Users\username\o3de\Code\Editor\ViewPane.cpp (222) : CLayoutViewPane::SetViewClass
<14:05:12> (System) - C:\Users\username\o3de\Code\Editor\LayoutWnd.cpp (207) : CLayoutWnd::MaximizeViewport
<14:05:12> (System) - C:\Users\username\o3de\Code\Editor\LayoutWnd.cpp (307) : CLayoutWnd::CreateLayout
<14:05:12> (System) - C:\Users\username\o3de\Code\Editor\MainWindow.cpp (442) : MainWindow::InitCentralWidget
<14:05:12> (System) - C:\Users\username\o3de\Code\Editor\MainWindow.cpp (454) : MainWindow::Initialize
<14:05:12> (System) - C:\Users\username\o3de\Code\Editor\CryEdit.cpp (1664) : CCryEditApp::InitInstance
<14:05:12> (System) - C:\Users\username\o3de\Code\Editor\CryEdit.cpp (4298) : CryEditMain
<14:05:12> (System) - C:\Users\username\o3de\Code\Editor\main.cpp (31) : main
<14:05:12> (System) - C:\qt_build\Lumberyard3rdPartyQt\qtbase\src\winmain\qtmain_win.cpp (97) : WinMain
<14:05:12> (System) - d:\agent\_work\4\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl (288) : __scrt_common_main_seh
<14:05:12> (System) - 00007FFC25AA7C24 (KERNEL32) : BaseThreadInitThunk
<14:05:12> (System) - 00007FFC27A4D721 (ntdll) : RtlUserThreadStart
<14:05:12> (System) - ==================================================================
<14:05:12> (System) - ====Assert added to ignore list by spec and verbosity setting.====
<14:05:13> () - 00007FFC253F0E4A (KERNELBASE) : UnhandledExceptionFilter
<14:05:13> () - 00007FFC27A858F2 (ntdll) : memset
<14:05:13> () - 00007FFC27A6D376 (ntdll) : _C_specific_handler
<14:05:13> () - 00007FFC27A81FFF (ntdll) : _chkstk
<14:05:13> () - 00007FFC27A4AAD9 (ntdll) : RtlRaiseException
<14:05:13> () - 00007FFC27A80C6E (ntdll) : KiUserExceptionDispatcher
<14:05:13> () - C:\Users\username\o3de\Gems\Atom\RPI\Code\Source\RPI.Public\Shader\Shader.cpp (432) : AZ::RPI::Shader::GetDrawListTag
<14:05:13> Successfully recorded DMP file:  'C:\Users\username\.o3de\Projects\NewProject\user\log/error.dmp'

**Reply by RogueRocketGames** [reply]
the commandline suggestion above (#2) seemed to net the same result.  i just attached the whole log to avoid confusion.  maybe the pak file missing is the issue?  i'm more than suspicious that i've done something wrong but its hard to tell what.  i did build the project through the launcher app as well.

[Editor-scrubbed.txt](https://github.com/o3de/o3de/files/6780093/Editor-scrubbed.txt)

**Reply by amznestebanpapp** [reply]
Looks like is trying to load the project: `C:\Users\username\.o3de\Projects\NewProject`, is this an empty project? Looks like it didnt load the minimal set of gems to run anything.
Check what you have in `C:\Users\username\.o3de\Projects\NewProject\Gem\Code\enabled_gems.cmake`. Did you build your project?

**Reply by RogueRocketGames** [reply]
yeah i just tried to make a new project and just left it as a name i was just trying to get the whole thing up.
i did build it (via the launcher tool, not via commandline).  it is empty, i have never been able to open it in the first place.  

Yes I built the project, I also just re-saved the gems config from the launcher and re-built again.  though, the gems cmake file you reference seems to live in a different sub-folder, is that a problem?

C:\Users\username\.o3de\Projects\NewProject\Code\enabled_gems.cmake
(no Gem subfolder in path)

the contents I assume are default:

set(ENABLED_GEMS
    NewProject
    Atom_AtomBridge
    AudioSystem
    AWSCore
    CameraFramework
    DebugDraw
    EditorPythonBindings
    EMotionFX
    GameState
    ImGui
    LandscapeCanvas
    LyShine
    Multiplayer
    PhysX
    PrimitiveAssets
    SaveData
    ScriptCanvasPhysics
    ScriptEvents
    StartingPointInput
    TextureAtlas
    WhiteBox
)

**Reply by amznestebanpapp** [reply]
yeah, the path should be ok.
Can you try to find a `cmake_dependencies.newproject.editor.setreg` file? that is the file that indicates what gems will be loaded.
It should be in a path like this one: `C:\Users\username.o3de\Projects\NewProject\build\windows_vs2019\bin\profile\Registry`

If you want to jump into discord, we can do this back-forward a bit more interactive.

**Reply by RogueRocketGames** [reply]
i was able to finally get it up!  i think i had several sneaky things slightly wrong that were easy to miss or miss were happening.  i'm unsure entirely how many of these were really an issue but:

- i had accidentally initially run the o3de launcher from the original DEVELOPMENT folder from the git repo location, not the o3de install folder as per instructions.  i can't tell for sure but i suspect it had caused some configuration configuration somewhere maybe
- somewhere in the docs i now can't find, it seemed to suggest the launcher build would properly register the engine/editor but i'm not sure it was true (or was incorrect due to prior bullet).  I manually re-registered engine per the instructions
- i nuked all prior projects and re-created a new one, rebuilt it.
- i discovered that the asset processor gets re-launched per project, and i had multiple running, but each had an error about multiple instances running and it was not processing assets.  it didn't proactively pop up the error dialog until i found it in the tray, so i had no idea it was happening
- i had to un-set a few engine folder locations.  in desperation i had tried pointing templates and gems to the install folder templates and gems folders, but this didn't help, and seemed wrong.  

overall there was definitely some user error, but it felt quite easy to make those mistakes.  but i seem to be finally past step 1:  get engine and editor running.  =)

**Reply by amznestebanpapp** [reply]
FYI: @lawsonamzn @lumberyard-employee-dm @AMZN-ScottR 
Another case of a someone running into registration issues. 

@RogueRocketGames yes, it seems that you registered an engine and then the Project Manager didnt register the new engine. 
Thanks for all those details, seems we have a couple of things to fix.
For future reference, the registration ends up in `%USERPROFILE%/.o3de`, so if you want to start registrations from scratch, nuke that folder.
